src/Categories/Category/Instance/Monoidals.agda

```diff
@@ -9,44 +9,38 @@ open import Categories.Category.Helper
 open import Categories.Category.Monoidal
 open import Categories.Functor.Monoidal
 open import Categories.Functor.Monoidal.Properties
-open import Categories.NaturalTransformation.NaturalIsomorphism
+open import Categories.NaturalTransformation.NaturalIsomorphism.Monoidal
 
 module _ o ℓ e where
 
   Monoidals : Category (suc (o ⊔ ℓ ⊔ e)) (o ⊔ ℓ ⊔ e) (o ⊔ ℓ ⊔ e)
   Monoidals = categoryHelper record
     { Obj       = MonoidalCategory o ℓ e
     ; _⇒_       = MonoidalFunctor
-    ; _≈_       = λ F G → M.F F ≃ M.F G
+    ; _≈_       = Lax._≃_
     ; id        = idF-Monoidal _
     ; _∘_       = ∘-Monoidal
-    ; assoc     = λ {_ _ _ _ F G H} → associator (M.F F) (M.F G) (M.F H)
-    ; identityˡ = unitorˡ
-    ; identityʳ = unitorʳ
-    ; equiv     = record
-      { refl  = ≃.refl
-      ; sym   = ≃.sym
-      ; trans = ≃.trans
-      }
+    -- NOTE: these η-expanded versions typecheck much faster...
+    ; assoc     = λ {_ _ _ _ F G H} → associator {F = F} {G} {H}
+    ; identityˡ = λ {_ _ F} → unitorˡ {F = F}
+    ; identityʳ = λ {_ _ F} → unitorʳ {F = F}
+    ; equiv     = isEquivalence
     ; ∘-resp-≈  = _ⓘₕ_
     }
-    where module M = MonoidalFunctor
+    where open Lax
 
   StrongMonoidals : Category (suc (o ⊔ ℓ ⊔ e)) (o ⊔ ℓ ⊔ e) (o ⊔ ℓ ⊔ e)
   StrongMonoidals = categoryHelper record
     { Obj       = MonoidalCategory o ℓ e
     ; _⇒_       = StrongMonoidalFunctor
-    ; _≈_       = λ F G → M.F F ≃ M.F G
+    ; _≈_       = Strong._≃_
     ; id        = idF-StrongMonoidal _
     ; _∘_       = ∘-StrongMonoidal
-    ; assoc     = λ {_ _ _ _ F G H} → associator (M.F F) (M.F G) (M.F H)
-    ; identityˡ = unitorˡ
-    ; identityʳ = unitorʳ
-    ; equiv     = record
-      { refl  = ≃.refl
-      ; sym   = ≃.sym
-      ; trans = ≃.trans
-      }
+    -- NOTE: these η-expanded versions typecheck much faster...
```

Comment:
I think it would be worth finding ways to document this issue and report it as an issue on the main agda repo. I'm not sure the principal devs are aware of this.  There might be nothing that can be done but without awareness, there is definitely no chance of a fix!

Reply:
Yep, I'll open an issue in the Agda repo.

I suspect, though, that the devs will tell me to craft an minimal example, which I won't be able to do (this only seems to happen when the development gets sufficiently complex).

Reply:
Reported as  agda/agda#5060.

Reply:
FYI, @andreasabel responded to the issue I opened at agda/agda#5060 a while ago. The issue has been closed now, so I just wanted to point it out in case you hadn't seen the response.